persistence/memcached.go

```diff
@@ -23,6 +23,11 @@ func (c *MemcachedStore) Set(key string, value interface{}, expires time.Duratio
 	return c.invoke((*memcache.Client).Set, key, value, expires)
 }
 
+// TODO: stub func
+func (c *MemcachedStore) MSetNX(expires time.Duration, kv ...interface{}) error {
```

Comment:
Same thing: why add this to the interface, especially if you're not implementing it in all the concrete types?

Reply:
I need to add those stub funcs into interface because of the way these funcs are exposed by go-cache, which is the one we directly import into our project.

For example, c.Cache.(persistence.CacheStore).Add in go-cache generic.go Add func, only CacheStore interface is exposed, not the underneath concrete type. If redis func is implemented in concrete type, then go-cache code also needs to be changed.

Reply:
I'm not sure that's correct.   see: https://github.com/BoseCorp/go-cache/blob/9d18a89ccd1b2e2480d512decb5958eb44e4b272/generic.go#L405 which references: `newValue, err = c.Cache.(*persistence.RedisStore).IncrementAtomic(key, n)` which is the persistent store behind the interface.

Reply:
You are correct. RedisStore is exposed. I will make the change accordingly.

Reply:
I have updated the code, removing stub func from common interface, inmemory and memcached, and moved the three redis-specific test cases to redis_test.go. Local test is successful:

go test -v
=== RUN   TestInMemoryCache_TypicalGetSet
--- PASS: TestInMemoryCache_TypicalGetSet (0.00s)
=== RUN   TestInMemoryCache_IncrDecr
--- PASS: TestInMemoryCache_IncrDecr (0.00s)
=== RUN   TestInMemoryCache_Expiration
--- PASS: TestInMemoryCache_Expiration (8.01s)
=== RUN   TestInMemoryCache_EmptyCache
--- PASS: TestInMemoryCache_EmptyCache (0.00s)
=== RUN   TestInMemoryCache_Replace
--- PASS: TestInMemoryCache_Replace (2.00s)
=== RUN   TestInMemoryCache_Add
--- PASS: TestInMemoryCache_Add (2.00s)
=== RUN   TestMemcachedBinary_TypicalGetSet
--- PASS: TestMemcachedBinary_TypicalGetSet (0.00s)
=== RUN   TestMemcachedBinary_IncrDecr
--- PASS: TestMemcachedBinary_IncrDecr (0.00s)
=== RUN   TestMemcachedBinary_Expiration
--- PASS: TestMemcachedBinary_Expiration (8.00s)
=== RUN   TestMemcachedBinary_EmptyCache
--- PASS: TestMemcachedBinary_EmptyCache (0.00s)
=== RUN   TestMemcachedBinary_Replace
--- PASS: TestMemcachedBinary_Replace (2.00s)
=== RUN   TestMemcachedBinary_Add
--- PASS: TestMemcachedBinary_Add (2.00s)
=== RUN   TestMemcachedBinaryWithConfig_TypicalGetSet
--- PASS: TestMemcachedBinaryWithConfig_TypicalGetSet (0.00s)
=== RUN   TestMemcachedBinaryWithConfig_IncrDecr
--- PASS: TestMemcachedBinaryWithConfig_IncrDecr (0.00s)
=== RUN   TestMemcachedBinaryWithConfig_Expiration
--- PASS: TestMemcachedBinaryWithConfig_Expiration (8.00s)
=== RUN   TestMemcachedBinaryWithConfig_EmptyCache
--- PASS: TestMemcachedBinaryWithConfig_EmptyCache (0.00s)
=== RUN   TestMemcachedBinaryWithConfig_Replace
--- PASS: TestMemcachedBinaryWithConfig_Replace (2.00s)
=== RUN   TestMemcachedBinaryWithConfig_Add
--- PASS: TestMemcachedBinaryWithConfig_Add (2.00s)
=== RUN   TestMemcachedCache_TypicalGetSet
--- PASS: TestMemcachedCache_TypicalGetSet (0.00s)
=== RUN   TestMemcachedCache_IncrDecr
--- PASS: TestMemcachedCache_IncrDecr (0.00s)
=== RUN   TestMemcachedCache_Expiration
--- PASS: TestMemcachedCache_Expiration (8.00s)
=== RUN   TestMemcachedCache_EmptyCache
--- PASS: TestMemcachedCache_EmptyCache (0.00s)
=== RUN   TestMemcachedCache_Replace
--- PASS: TestMemcachedCache_Replace (2.00s)
=== RUN   TestMemcachedCache_Add
--- PASS: TestMemcachedCache_Add (2.00s)
=== RUN   TestRedisCache_MgetTwoKeys
--- PASS: TestRedisCache_MgetTwoKeys (0.00s)
    redis_test.go:109: test1: foo1, test2: foo2, err: <nil>
=== RUN   TestRedisCache_MsetNXTwoKeys
--- PASS: TestRedisCache_MsetNXTwoKeys (0.00s)
=== RUN   TestRedisCache_MsetNXThenMgetThreeKeys
--- PASS: TestRedisCache_MsetNXThenMgetThreeKeys (0.00s)
=== RUN   TestRedisCache_TypicalGetSet
--- PASS: TestRedisCache_TypicalGetSet (0.00s)
=== RUN   TestRedisCache_IncrDecr
--- PASS: TestRedisCache_IncrDecr (0.00s)
=== RUN   TestRedis_IncrAtomic
--- PASS: TestRedis_IncrAtomic (2.00s)
=== RUN   TestRedisCache_Expiration
--- PASS: TestRedisCache_Expiration (8.01s)
=== RUN   TestRedisCache_EmptyCache
--- PASS: TestRedisCache_EmptyCache (0.00s)
=== RUN   TestRedisCache_Replace
--- PASS: TestRedisCache_Replace (2.00s)
=== RUN   TestRedisCache_Add
--- PASS: TestRedisCache_Add (2.01s)
PASS
ok  	github.com/BoseCorp/cache/persistence	62.126s